bundles/org.eclipse.core.jobs/src/org/eclipse/core/runtime/jobs/ProgressProvider.java

```diff
@@ -10,6 +10,7 @@
  *
  * Contributors:
  *     IBM Corporation - initial API and implementation
+ *     Christoph Läubrich - Issue #40 - UI freezes when a ThreadJob is waiting to run in the UI Thread
```

Comment:
```suggestion
```

nitpicking just to test the suggestion button: according to vogellas comment this line should not be added too.

Reply:
cool, i did not know this function. thanks for sharing the info. is it now me or you who should press "commit suggestion"?

Reply:
I have enabled "allow edit by maintainers" so you could do that as well, but normally the PR author would pick the ones he find suitable, edit the file manually or could further discuss the issue.

If there are multiple one, one could also add them to a batch to use multiple at once (if there are any).

Reply:
Done